synapse/storage/events.py

```diff
@@ -515,16 +515,21 @@ def _get_new_state_after_events(self, events_context, new_latest_event_ids):
                     if ctx.current_state_ids is None:
                         raise Exception("Unknown current state")
 
+                    if ctx.state_group is None:
+                        # I don't think this can happen, but let's double-check
+                        raise Exception(
+                            "Context for new extremity event %s has no state "
+                            "group" % event_id,
```

Comment:
Nit: I do prefer always using tuples rather than relying on strings working.

Reply:
as in you'd prefer `(event_id, )` ?

Reply:
Generally yes, (or `(event_id,)` without the space), as its very confusing if there is a bug and `event_id` suddenly becomes iterable and then the logging explodes. In this particular case its probably fine though

Reply:
fair